data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostProposerSlashing.java

```diff
@@ -70,24 +90,22 @@ public PostProposerSlashing(final NodeDataProvider provider, final JsonProvider
       })
   @Override
   public void handle(final Context ctx) throws Exception {
-    try {
-      final ProposerSlashing proposerSlashing =
-          parseRequestBody(ctx.body(), ProposerSlashing.class);
-      InternalValidationResult result =
-          nodeDataProvider.postProposerSlashing(proposerSlashing).join();
-      if (result.code().equals(ValidationResultCode.IGNORE)
-          || result.code().equals(ValidationResultCode.REJECT)) {
-        ctx.status(SC_BAD_REQUEST);
-        ctx.json(
-            BadRequest.badRequest(
-                jsonProvider,
-                "Invalid proposer slashing, it will never pass validation so it's rejected"));
-      } else {
-        ctx.status(SC_OK);
-      }
-    } catch (final IllegalArgumentException | JsonMappingException e) {
-      ctx.json(BadRequest.badRequest(jsonProvider, e.getMessage()));
-      ctx.status(SC_BAD_REQUEST);
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final ProposerSlashing proposerSlashing = request.getRequestBody();
+    final InternalValidationResult result =
+        nodeDataProvider.postProposerSlashing(proposerSlashing).join();
```

Comment:
surprised we have any like this.
`.join` is because we're waiting for a safe future to complete, this should be following the asyncRequest model and the rest of the logic inside `.thenApply`...